File: flutter_app1/android/app/src/main/kotlin/com/example/flutter_app1/CustomFlutterView.kt
package com.example.flutter_app1

import android.view.View

class CustomFlutterView(context: Context, messenger: BinaryMessenger, viewId: Int, args: Map<String, Any>?) : PlatformView, MethodChannel.MethodCallHandler {

    val textView: TextView = TextView(context)
    private var methodChannel: MethodChannel

    init {
        args?.also {
            textView.text = it["text"] as String
        }
        methodChannel = MethodChannel(messenger, "com.flutter.guide.MyFlutterView")
        methodChannel.setMethodCallHandler(this)
    }

     fun getView(): View {

        return textView
    }

    fun dispose() {
        methodChannel.setMethodCallHandler(null)
    }

     fun onMethodCall(call: MethodCall, result: MethodChannel.Result) {
        if (call.method == "setText") {
            val name = call.argument("name") as String?
            val age = call.argument("age") as Int?
            textView.text = "hello,$name,年龄：$age"
        } else if (call.method == "getData") {
            val name = call.argument("name") as String?
            val age = call.argument("age") as Int?

            var map = mapOf("name" to "hello,$name",
                    "age" to "$age"
            )
            result.success(map)
        } else {
            result.notImplemented()
        }
    }

}

File: flutter_app1/ios/Runner/CustomFlutterView.swift
//
//  CustomFlutterView.swift
//  Runner
//
//  Created by ksmacmini on 2021/12/2.
//

import Foundation
import Flutter

class CustomFlutterView: NSObject,FlutterPlatformView {
    
    let label = UILabel()
    var age:Int = 0
        

    init(_ frame: CGRect,viewID: Int64,args :Any?,messenger :FlutterBinaryMessenger) {
            super.init()
            if(args is NSDictionary){
                let dict = args as! NSDictionary
                self.label.text  = dict.value(forKey: "text") as! String
            }else{

            self.label.text = "test-test"
            }

            let methodChannel = FlutterMethodChannel(name: "com.flutter.guide.FlutterView_\(viewID)", binaryMessenger: messenger)
            methodChannel.setMethodCallHandler { (call, result) in
              
                
                let method:String = call.method
                if(method == "getData") {
                    
                    result(["message":"我是原生数据\(self.age)"]);
                    
                        
                }
                
                if (method == "setText") {
            
                    if let dict = call.arguments as? Dictionary<String, Any> {
                        let name:String = dict["name"] as? String ?? ""
                        let age:Int = dict["age"] as? Int ?? -1
                        self.age = age
                        self.label.text = "hello,\(name),年龄：\(age)"
                        
                    }
              
                }
            }
        }
        
        func view() -> UIView {
            return label
        }  
    
}
